src/sm/main/full_tracer.cpp

```diff
@@ -466,12 +468,16 @@ void FullTracer::onOpcode (Context &ctx, const RomCommand &cmd)
     {
         Fea lenMemValue = ctx.mem[offsetCtx + lengthMemOffset];
         fea2scalar(ctx.fr, auxScalar, lenMemValue.fe0, lenMemValue.fe1, lenMemValue.fe2, lenMemValue.fe3, lenMemValue.fe4, lenMemValue.fe5, lenMemValue.fe6, lenMemValue.fe7);
-        lenMemValueFinal = auxScalar.get_ui();
+        lenMemValueFinal = ceil(double(auxScalar.get_ui())/32);
```

Comment:
be careful with floating point numbers, it's better to avoid them, especially with big numbers (int64). It's better something  like (x >> 5) + ((x & 0x1F != 0)